src/php/ext/grpc/channel.c

```diff
@@ -222,10 +408,46 @@ PHP_METHOD(Channel, watchConnectivityState) {
  */
 PHP_METHOD(Channel, close) {
   wrapped_grpc_channel *channel = Z_WRAPPED_GRPC_CHANNEL_P(getThis());
-  if (channel->wrapped != NULL) {
-    grpc_channel_destroy(channel->wrapped);
-    channel->wrapped = NULL;
+  gpr_mu_lock(&channel->wrapper->mu);
+  if (channel->wrapper->wrapped != NULL) {
+    grpc_channel_destroy(channel->wrapper->wrapped);
+    channel->wrapper->wrapped = NULL;
+  }
+  gpr_mu_unlock(&channel->wrapper->mu);
+
+  php_grpc_delete_persistent_list_entry(channel->wrapper->key,
+                                        strlen(channel->wrapper->key)
+                                        TSRMLS_CC);
+}
+
+// Delete an entry from the persistent list
+// Note: this does not destroy or close the underlying grpc_channel
+void php_grpc_delete_persistent_list_entry(char *key, php_grpc_int key_len
+                                           TSRMLS_DC) {
+  php_grpc_zend_resource *rsrc;
+  if (PHP_GRPC_PERSISTENT_LIST_FIND(&EG(persistent_list), key,
```

Comment:
You might want another mutex around this global list, it could be accessed by multiple threads and I'm guessing its not threadsafe.